clients/src/test/java/org/apache/kafka/common/security/oauthbearer/internals/secured/RefreshingHttpsJwksTest.java

```diff
@@ -195,4 +231,68 @@ public String getBody() {
         return Mockito.spy(httpsJwks);
     }
 
+    /**
+     * A mock ScheduledExecutorService just for the test. Note that this is not a generally reusable mock as it does not
+     * implement some interfaces like scheduleWithFixedDelay, etc. And it does not return ScheduledFuture correctly.
+     */
+    private class MockExecutorService implements MockTime.Listener {
+        private final MockTime time;
+
+        private final TreeMap<Long, List<AbstractMap.SimpleEntry<Long, KafkaFutureImpl<Long>>>> waiters = new TreeMap<>();
+
+        public MockExecutorService(MockTime time) {
+            this.time = time;
+            time.addListener(this);
+        }
+
+        @Override
+        public synchronized void onTimeUpdated() {
```

Comment:
Could you add some java doc on what we're trying to do in this method? Same as below. Thanks.

Reply:
Done

Reply:
This is a test-only class so I think it should be fine? We should make sure scheduled thing does not have exceptions?